src/Cooked/MockChain/Testing.hs

```diff
@@ -295,10 +133,175 @@ assertSubset l r =
         l
     )
 
+-- | Asserts whether 2 sets are equal, both given as lists.
 assertSameSets :: (Show a, Eq a) => [a] -> [a] -> HU.Assertion
 assertSameSets l r =
   HU.assertBool
     ("expected lists of the same length, got " ++ show (length l) ++ " and " ++ show (length r))
     (length l == length r)
     .&&. assertSubset l r
     .&&. assertSubset r l
+
+-- * Testing mockchain runs
+
+-- | Data structure to test a mockchain run
+data Test a prop = Test
+  { -- | The mockchain run to test
+    testRun :: StagedMockChain a,
```

Comment:
Don't we usually call those "traces" instead of "runs"?
To me, "trace" is the data describing a scenario, and "run" is its execution / result.

Reply:
I will change the naming